arangod/RestHandler/RestMetricsHandler.cpp

```diff
@@ -79,23 +85,41 @@ RestStatus RestMetricsHandler::execute() {
   }
 
   if (_request->requestType() != RequestType::GET) {
+    // TODO(MBkkt) Now our API return 405 errorCode for 400 HTTP response code
+    //             I think we should fix it, but its breaking change
     generateError(ResponseCode::BAD, TRI_ERROR_HTTP_METHOD_NOT_ALLOWED);
     return RestStatus::DONE;
   }
 
-  bool foundServerIdParameter;
-  auto const& serverId = _request->value("serverId", foundServerIdParameter);
-
-  if (ServerState::instance()->isCoordinator() && foundServerIdParameter) {
-    if (serverId != ServerState::instance()->getId()) {
-      // not ourselves! - need to pass through the request
-      auto& ci = server().getFeature<ClusterFeature>().clusterInfo();
-      if (!ci.serverExists(serverId)) {
-        generateError(rest::ResponseCode::NOT_FOUND,
-                      TRI_ERROR_HTTP_BAD_PARAMETER,
-                      std::string("unknown serverId supplied."));
-        return RestStatus::DONE;
-      }
+  bool foundServerId;
+  bool foundType;
+  bool foundMode;
+  auto const& serverId = _request->value("serverId", foundServerId);
+  std::string_view type = _request->value("type", foundType);
+  std::string_view mode = _request->value("mode", foundMode);
+
+  foundServerId = foundServerId && ServerState::instance()->isCoordinator() &&
+                  serverId != ServerState::instance()->getId();
+  // TODO(MBkkt) I think in the future we should return an error
+  //             if the ServerId is not a Coordinator or it's our ServerId.
+  //             But now it will be breaking changes.
+
+  std::string error;
+  size_t notFound = 0;
```

Comment:
Why not use just boolean flag? In theory message ext could be changed and accidentally size of "Not Found" error text will match one of  "Wrong parameter" error text.